src/net.cpp

```diff
@@ -674,6 +674,18 @@ int V1TransportDeserializer::readData(const char *pch, unsigned int nBytes)
     memcpy(&vRecv[nDataPos], pch, nCopy);
     nDataPos += nCopy;
 
+    if (Complete()) {
```

Comment:
It may be good to add a comment here, along the lines of
```diff
     if (Complete()) {
+        // Verify message has valid checksum. If invalid, reject and disconnect the peer.
```